pkg/hubble/exporter/dynamic_exporter.go

```diff
@@ -0,0 +1,207 @@
+	if lock {
+		d.mutex.RLock()
+		defer d.mutex.RUnlock()
+	}
+
+	var activeExporters, inactiveExporters float64
+
+	for name, me := range d.managedExporters {
+		var value float64
+		if me.config.End.After(time.Now()) {
+			value = 1
+			activeExporters++
+		} else {
+			value = 0
+			inactiveExporters++
+		}
+		metrics.IndividualExportersStatus.WithLabelValues(name).Set(value)
+	}
+
+	metrics.Exporters.WithLabelValues("active").Set(activeExporters)
+	metrics.Exporters.WithLabelValues("inactive").Set(inactiveExporters)
+}
+
+func (d *dynamicExporter) onConfigReload(ctx context.Context, config dynamicExportersConfig) {
+	d.mutex.Lock()
+	defer d.mutex.Unlock()
+
+	configuredFlowLogNames := make(map[string]bool)
+	for _, flowlog := range config.FlowLogs {
+		configuredFlowLogNames[flowlog.Name] = true
+		if _, ok := d.managedExporters[flowlog.Name]; ok {
+			if d.applyUpdatedConfig(ctx, flowlog) {
+				metrics.ExporterReconfigurations.WithLabelValues("update").Inc()
+			}
+		} else {
+			d.applyNewConfig(ctx, flowlog)
+			metrics.ExporterReconfigurations.WithLabelValues("add").Inc()
+		}
+	}
+
+	for flowLogName := range d.managedExporters {
+		if _, ok := configuredFlowLogNames[flowLogName]; !ok {
+			d.applyRemovedConfig(flowLogName)
+			metrics.ExporterReconfigurations.WithLabelValues("remove").Inc()
+		}
+	}
+
+	d.reportGaugeMetrics(false)
+	d.updateLastAppliedConfigGauges(config)
+}
+
+func (d *dynamicExporter) applyNewConfig(ctx context.Context, flowlog *flowLogConfig) {
+	exporterOpts := []exporteroption.Option{
+		exporteroption.WithPath(flowlog.FilePath),
+		exporteroption.WithMaxSizeMB(d.maxFileSizeMB),
+		exporteroption.WithMaxBackups(d.maxBackups),
+		exporteroption.WithAllowList(flowlog.IncludeFilters),
+		exporteroption.WithDenyList(flowlog.ExcludeFilters),
+		exporteroption.WithFieldMask(flowlog.FieldMask),
+	}
+
+	exporter, err := NewExporter(ctx, d.logger.WithField("flowLogName", flowlog.Name), exporterOpts...)
+	if err != nil {
+		d.logger.Errorf("Failed applying flowlog for name: %s; %v", flowlog.Name, err)
+	}
+
+	d.managedExporters[flowlog.Name] = &managedExporter{
+		config:   flowlog,
+		exporter: exporter,
+	}
+
+}
+
+func (d *dynamicExporter) applyUpdatedConfig(ctx context.Context, flowlog *flowLogConfig) bool {
+	m := d.managedExporters[flowlog.Name]
+	if m.config.equals(flowlog) {
+		return false
+	}
+	d.applyRemovedConfig(flowlog.Name)
+	d.applyNewConfig(ctx, flowlog)
+	return true
+}
+
+func (d *dynamicExporter) applyRemovedConfig(name string) {
+	m := d.managedExporters[name]
+	if err := m.exporter.Stop(); err != nil {
+		d.logger.Errorf("error stopping exporter %v", err)
+	}
+	delete(d.managedExporters, name)
+	//remove metric series for removed exporter
+	metrics.IndividualExportersStatus.DeleteLabelValues(name)
+}
+
+func (d *dynamicExporter) updateLastAppliedConfigGauges(config dynamicExportersConfig) {
+	hash, err := hashstructure.Hash(config, hashstructure.FormatV2, nil)
```

Comment:
Can we just hash the raw `[]byte`? I don't think we need to add an extra dependency here for hashing the data structure after parsing.

Reply:
fixed